message.go

```diff
@@ -139,24 +138,27 @@ func (m *Message) WriteTo(w io.Writer) (int64, error) {
 	return total, nil
 }
 
-// DecodeMessage deseralizes data (as []byte) and creates a new Message
+// DecodeMessage deserializes data (as []byte) and creates a new Message
+// message format:
+// [x][x][x][x][x][x][x][x][x][x][x][x][x][x][x][x][x][x][x][x][x][x][x][x][x][x][x][x][x][x]...
+// |       (int64)        ||    ||      (hex string encoded in ASCII)           || (binary)
```

Comment:
Docstring nitpick: `message ID` is binary w/r/t the protocol.  Practically we chose to use ascii printable characters for now.